### PR TITLE
Fix: estimator getter and lagged_label_name

### LINKED ISSUE #2217: Order of estimators for a multivariate target series with multi_models=True
Hi there,

I was wondering about the order in which estimators are stored, specifically, in an XGBoost model. Let's say, I train a model called `xgb` on a multivariate target series with two components `A` and `B`, and an `output_chunk_length=2` with `multi_models=True`. 
Are the estimators in `xgb.model.estimators_` stored as `[A1, A2, B1, B2]` or `[A1, B1, A2, B2]` (if e.g. `B2` donates the estimator for component `B` with horizon `2`)?

Alternatively, is there maybe something similar to `xgb.lagged_feature_names` to retrieve the "target names" of all estimators?

Thank you!

### PR DESCRIPTION
**Pull request by madtoinou**
Fixes #2217.

### Summary

- Fixed bug in `get_multioutput_estimator()` where the incorrect estimator could be returned
- Added an attribute to store the lagged label, each one corresponding to an estimator

### Other Information

- Added tests where the estimator are over-fitted on crafted examples to check if the appropriate estimator is returned by the getter.

### CODE REVIEW

File: darts/tests/models/forecasting/test_regression_models.py [hunk 1]
```diff
@@ -1270,57 +1270,115 @@ def test_historical_forecast(self, mode):
...
-            models += [
-                CatBoostModel(lags=lags, output_chunk_length=1, multi_models=True),
-                CatBoostModel(lags=lags, output_chunk_length=1, multi_models=False),
-                CatBoostModel(lags=lags, output_chunk_length=2, multi_models=True),
-                CatBoostModel(lags=lags, output_chunk_length=2, multi_models=False),
-            ]
+    @pytest.mark.parametrize(
+        "config", itertools.product(model_configs, [1, 2], [True, False])
+    )
+    def test_multioutput_validation(self, config):
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_regression_models.py:1290]: nice 👍

File: darts/tests/models/forecasting/test_regression_models.py [hunk 2]
```diff
@@ -1270,57 +1270,115 @@ def test_historical_forecast(self, mode):
...
+            for i in range(ocl):
+                for j in range(ts.width):
+                    sub_model = m.get_multioutput_estimator(horizon=i, target_dim=j)
+                    pred = sub_model.predict(dummy_feats)[0]
+                    # sub-model is overfitted on the training series
+                    assert np.abs(ts.width * i + j + 1 - pred) < 1e-2
+
+        # univariate, one-sub model per step in output_chunk_length
+        ocl = 3
+        ts = TimeSeries.from_values(np.array([0, 0, 0, 1, 2, 3, 4]).T)
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_regression_models.py:1322]: we can take one away at the end
```suggestion
        ts = TimeSeries.from_values(np.array([0, 0, 0, 0, 1, 2]).T)
```

File: darts/tests/models/forecasting/test_regression_models.py [hunk 3]
```diff
@@ -1270,57 +1270,115 @@ def test_historical_forecast(self, mode):
...
+        if model.output_chunk_length > 1 and model.multi_models:
+            assert isinstance(model.model, MultiOutputRegressor)
+        else:
+            assert not isinstance(model.model, MultiOutputRegressor)
 
-        for model in models:
-            model.fit(series=train, val_series=val)
-            if model.output_chunk_length > 1 and model.multi_models:
-                assert isinstance(model.model, MultiOutputRegressor)
+    def test_get_multioutput_estimator_multi_models(self):
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_regression_models.py:1303]: can be a bit tricky to understand.

I will add some suggestions which might help for interpretability

File: darts/tests/models/forecasting/test_regression_models.py [hunk 4]
```diff
@@ -1270,57 +1270,115 @@ def test_historical_forecast(self, mode):
...
+
+            assert len(m.model.estimators_) == ocl * ts.width
+
+            dummy_feats = np.array([[0, 0, 0] * ts.width])
+            for i in range(ocl):
+                for j in range(ts.width):
+                    sub_model = m.get_multioutput_estimator(horizon=i, target_dim=j)
+                    pred = sub_model.predict(dummy_feats)[0]
+                    # sub-model is overfitted on the training series
+                    assert np.abs(ts.width * i + j + 1 - pred) < 1e-2
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_regression_models.py:1318]: If we adapt the training examples a bit further below, we can simply check that the predictions are identical to the estimator counter
```suggestion
            estimator_counter = 0
            for i in range(ocl):
                for j in range(ts.width):
                    sub_model = m.get_multioutput_estimator(horizon=i, target_dim=j)
                    pred = sub_model.predict(dummy_feats)[0]
                    # sub-model is overfitted on the training series
                    assert np.abs(estimator_counter - pred) < 1e-2
                    estimator_counter += 1
```

File: darts/tests/models/forecasting/test_regression_models.py [hunk 5]
```diff
@@ -1270,57 +1270,115 @@ def test_historical_forecast(self, mode):
...
+        # estimators_[2] labels : [3, 3]
+        helper_check_overfitted_estimators(ts, ocl)
+
+        # multivariate, one sub-model per position, per component
+        ocl = 2
+        ts = TimeSeries.from_values(
+            np.array(
+                [
+                    [0, 0, 0, 0, 2, 4],
+                    [0, 0, 0, 0, 4, 4],
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_regression_models.py:1344]:
```suggestion
                    [0, 0, 0, 0, 2],
                    [0, 0, 0, 1, 3],
```

File: darts/tests/models/forecasting/test_regression_models.py [hunk 6]
```diff
@@ -1270,57 +1270,115 @@ def test_historical_forecast(self, mode):
...
+                ]
+            ).T
+        )
+        # estimators_[0] labels : [0, 2]
+        # estimators_[1] labels : [0, 4]
+        # estimators_[2] labels : [2, 4]
+        # estimators_[3] labels : [4, 4]
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_regression_models.py:1351]:
```suggestion
        # estimators_[0] labels : [0]
        # estimators_[1] labels : [1]
        # estimators_[2] labels : [2]
        # estimators_[3] labels : [3]
```

File: darts/tests/models/forecasting/test_regression_models.py [hunk 7]
```diff
@@ -1270,57 +1270,115 @@ def test_historical_forecast(self, mode):
...
+        helper_check_overfitted_estimators(ts, ocl)
+
+        # multivariate, one sub-model per component
+        ocl = 1
+        ts = TimeSeries.from_values(
+            np.array([[0, 0, 0, 1, 1], [0, 0, 0, 2, 2], [0, 0, 0, 3, 3]]).T
+        )
+        # estimators_[0] labels : [1, 1]
+        # estimators_[1] labels : [2, 2]
+        # estimators_[2] labels : [3, 3]
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_regression_models.py:1335]:
```suggestion
        # estimators_[0] labels : [1]
        # estimators_[1] labels : [2]
        # estimators_[2] labels : [3]
```

File: darts/tests/models/forecasting/test_regression_models.py [hunk 8]
```diff
@@ -1270,57 +1270,115 @@ def test_historical_forecast(self, mode):
...
+                    pred = sub_model.predict(dummy_feats)[0]
+                    # sub-model is overfitted on the training series
+                    assert np.abs(ts.width * i + j + 1 - pred) < 1e-2
+
+        # univariate, one-sub model per step in output_chunk_length
+        ocl = 3
+        ts = TimeSeries.from_values(np.array([0, 0, 0, 1, 2, 3, 4]).T)
+        # estimators_[0] labels : [1, 2]
+        # estimators_[1] labels : [2, 3]
+        # estimators_[2] labels : [3, 4]
```
Comment (dennisbader) [inline darts/tests/models/forecasting/test_regression_models.py:1325]:
```suggestion
        # estimators_[0] labels : [1]
        # estimators_[1] labels : [2]
        # estimators_[2] labels : [3]
```

**Review by dennisbader — APPROVED** [review]
Great PR, thanks @madtoinou 🚀
